File: Java/lab4/src/main/java/org/example/factory/PhoneFactory.java
package org.example.factory;

import org.example.interfaces.Factory;
import org.example.interfaces.LaptopProduct;
import org.example.interfaces.PcProduct;
import org.example.interfaces.PhoneProduct;
import org.example.phone.Iphone;
import org.example.phone.Pixel;
import org.example.phone.Samsung;
public class PhoneFactory implements Factory {
    @Override
    public LaptopProduct createLaptop(String typeOfLaptop) {
        return null;
    }
    @Override
    public PcProduct createPc(String typeOfPc) {
        return null;
    }
    public PhoneProduct createPhone(String typeOfPhone) {
        return switch (typeOfPhone) {
            case "Samsung" -> new Samsung();
            case "Iphone" -> new Iphone();
            case "Pixel" -> new Pixel();
            default -> null;
        };
    }
}

File: Python/lab4/factory/PhoneFactory.py
from interfaces.PhoneProduct import PhoneProduct
from interfaces.Factory import Factory
from phone.Iphone import Iphone
from phone.Pixel import Pixel
from phone.Samsung import Samsung
from interfaces.LaptopProduct import LaptopProduct
from interfaces.PcProduct import PcProduct

class PhoneFactory(Factory):
    def createPhone(self, type_of_phone) -> PhoneProduct:
        if type_of_phone == 'Samsung':
            return Samsung()
        elif type_of_phone == "Pixel":
            return Pixel()
        elif type_of_phone == "Iphone":
            return Iphone()
        else:
            return None

    def createPc(self, type_of_pc) -> PcProduct:
        return None

    def createLaptop(self, type_of_pc) -> LaptopProduct:
        return None
